### PR TITLE
sig-autoscaling: add teams for karpenter repo

### PR DESCRIPTION
**Pull request by Priyankasaggu11929**
Ref: https://github.com/kubernetes/org/issues/4562

/assign @kubernetes/owners

### CODE REVIEW

**Comment by Priyankasaggu11929** [comment]
`pull-org-test-all` is failing with following error:

```
config_test.go:268: The following members of team karpenter-admins are not Kubernetes SIGs org members: gjtempleton, maciekpytel
config_test.go:268: The following members of team sig-autoscaling-leads are not Kubernetes SIGs org members: gjtempleton
config_test.go:268: The following members of team karpenter-maintainers are not Kubernetes SIGs org members: gjtempleton, maciekpytel 
```
    
Both @gjtempleton and @MaciekPytel are existing members of the kubernetes org.

Adding them to the `kubernetes-sigs` org as well to fix the failing CI test.

**Comment by gjtempleton** [comment]
Raised #4598 to add myself, will add maciekpytel to the same PR.

**Comment by Priyankasaggu11929** [comment]
Hello @gjtempleton, I've added you and @MaciekPytel to `kubernetes-sigs` org as part of this PR itself, so org membership requirements are fine now. No need for separate PRs.

**Comment by Priyankasaggu11929** [comment]
@MadhavJivrajani, @palnabarun need review/approval. Thanks!

File: config/kubernetes-sigs/sig-autoscaling/teams.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+teams:
```
Comment (jonathan-innis) [inline config/kubernetes-sigs/sig-autoscaling/teams.yaml:1]: Is there a way to create a team for the reviewers or is that solely handled by the `OWNERS_ALIASES` file that's in `kubernetes-sigs/karpenter`?

Comment (Priyankasaggu11929) [reply config/kubernetes-sigs/sig-autoscaling/teams.yaml:1]: Yea, creating a team would be fine.

Who all you need to be added to the reviewers team? And with what level of access (read is enough?)

Comment (jonathan-innis) [reply config/kubernetes-sigs/sig-autoscaling/teams.yaml:1]: Read should be enough. Just talking about reviewer-level. This is the list of users from the proposed PR: https://github.com/jonathan-innis/karpenter-core/blob/add-new-member-approvers/OWNERS_ALIASES#L14

Comment (Priyankasaggu11929) [reply config/kubernetes-sigs/sig-autoscaling/teams.yaml:1]: added `karpenter-reviewers` team

**Pull request merged**